knock

```diff
@@ -30,22 +30,33 @@ class Knocker(object):
         self.timeout = args.timeout / 1000
         self.delay = args.delay / 1000
         self.use_udp = args.udp
-        self.host = args.host
-        self.is_ipv6 = args.host.find(':') != -1
         self.ports = args.ports
 
+        self.address_family, self.socket_type, _, _, socket_address = socket.getaddrinfo(
+                host=args.host,
+                port=None,
+                type=0,
+                proto=socket.IPPROTO_UDP if self.use_udp else socket.IPPROTO_TCP,
```

Comment:
The type and proto inputs to `socket.getaddrinfo()` seem to emerg as items 2 and 3 in the tuple. Not sure they affect anything else, but the call to `socket.socket()` need the **SOCK_DGRAM** or **SOCK_STREAM** to create the right type. As you passed in type=0 to `socket.getaddrino()`, that was arriving at `socket.socket()` and like creating a SOCK_STREAM by default,

BTW, I'm testing on Windows 10 Python 3.5.2 against a CentOS6 linux and using tcpdump to verify the packet arrival.

My server has 3 types of addresses, IPv4, IPv6 and 6to4 (a transition variant of IPv6):
1. ipv4.netconfigs.com
2. ipv6.netconfigs.com
3. 6to4.netconfigs.com